### PR TITLE
Clean up runend metadata & stats

### PR DESCRIPTION
**Pull request by lwwmanning**

### CODE REVIEW

File: encodings/runend/src/compress.rs [hunk 1]
```diff
@@ -16,27 +17,56 @@ pub fn runend_encode(array: &PrimitiveArray) -> (PrimitiveArray, PrimitiveArray)
...
+    compressed_ends
+        .statistics()
+        .set(Stat::IsSorted, true.into());
+    compressed_ends
+        .statistics()
+        .set(Stat::IsStrictSorted, true.into());
+    if !compressed_ends.is_empty() {
+        compressed_ends
+            .statistics()
+            .set(Stat::Min, scalar_at(&compressed_ends, 0).vortex_unwrap());
```
Comment (robert3005) [inline encodings/runend/src/compress.rs:61]:
```suggestion
            .set(Stat::Min, 0.into());
```

File: encodings/runend/src/compress.rs [hunk 2]
```diff
@@ -16,27 +17,56 @@ pub fn runend_encode(array: &PrimitiveArray) -> (PrimitiveArray, PrimitiveArray)
...
+        compressed_ends.statistics().set(
+            Stat::Max,
+            scalar_at(&compressed_ends, compressed_ends.len() - 1).vortex_unwrap(),
```
Comment (robert3005) [inline encodings/runend/src/compress.rs:64]:
```suggestion
            array.len().into(),
```

File: encodings/runend/src/compress.rs (hunk 1 above)
Comment (lwwmanning) [reply encodings/runend/src/compress.rs:61]: this isn't necessarily always 0 if we slice?

File: encodings/runend/src/compress.rs (hunk 2 above)
Comment (lwwmanning) [reply encodings/runend/src/compress.rs:64]: again, slices & offsets maybe get us here?

File: encodings/runend/src/compress.rs (hunk 1 above)
Comment (robert3005) [reply encodings/runend/src/compress.rs:61]: But we just created this array, in general yes but even for a sliced array RunEnd encoding will reset the offset to 0

**Review by danking — APPROVED** [review]

File: encodings/runend/src/compress.rs (hunk 2 above)
Comment (robert3005) [reply encodings/runend/src/compress.rs:64]: Sooo what about this comment?

**Pull request merged**